src/sdmon.c

```diff
@@ -252,6 +252,75 @@ int main(int argc, const char *argv[]) {
     exit(0);
   }
 
+  //try transcend argument
+  cmd56_arg = 0x110005f9;
+  ret = CMD56_data_in(fd, cmd56_arg, data_in);
+  // we assume success when the call was successful AND the signature is not 0xff 0xff
+  if (ret == 0 && !((data_in[0] == 0xff && data_in[1] == 0xff) || (data_in[0] == 0x00 && data_in[1] == 0x00))) {
+	printf("\"signature\":\"0x%x 0x%x\",\n", data_in[0], data_in[1]);
+	if (data_in[0] == 0x54 && data_in[1] == 0x72) {
+		printf("\"Transcend\":\"true\",\n");
+		printf("\"Secured mode\": %d,\n", (int)(data_in[11]));
+	switch (data_in[16])
+	{
+		case 0x00:
+			printf("\"Bus width\": 1 bit\n");
+			break;
+		case 0x10:
+			printf("\"Bus width\": 4 bits\n");
+			break;
+	}
+	switch (data_in[18])
+	{
+		case 0x00:
+			printf("\"Speed mode\": Class 0\n");
+			break;
+		case 0x01:
+			printf("\"Speed mode\": Class 2\n");
+			break;
+		case 0x02:
+			printf("\"Speed mode\": Class 4\n");
+			break;
+		case 0x03:
+			printf("\"Speed mode\": Class 6\n");
+			break;
+		case 0x04:
+			printf("\"Speed mode\": Class 10\n");
+			break;
+	}
+	switch (data_in[19])
+	{
+		case 0x00:
+			printf("\"UHS speed grade\": Less than 10MB/s\n");
+			break;
+		case 0x01:
+			printf("\"UHS speed grade\": 10MB/s and higher\n");
+			break;
+		case 0x03:
+			printf("\"UHS speed grade\": 30MB/s and higher\n");
+			break;
+	}
+	printf("\"New bad blocks cnt\": 0x%02x,\n", data_in[26]);
+	printf("\"Runtime spare blocks cnt\": 0x%02x,\n", data_in[27]);
+	printf("\"Abnormal power loss\": %ld,\n", (long)((data_in[31] << 24) + (data_in[30] << 16) + (data_in[29] << 8) + data_in[28]));
+	printf("\"Minimum erase cnt\": %d,\n", (long)((data_in[35] << 24) + (data_in[34] << 16) + (data_in[33] << 8) + data_in[32]));
+	printf("\"Maximum erase cnt\": %d,\n", (long)((data_in[39]) + (data_in[38]) + (data_in[37]) + data_in[36]));
```

Comment:
don't we need bit shifts here?

Reply:
I think they should be here too, so I've made this small fix in the lates development release...
@J-Twin can you test this?

Reply:
Sure no problem! It is your project so you can modify the code. Like a previous fix you had applied, this broke printing too as you can see below:

sudo ./sdmon /dev/mmcblk0
{
"version": "v0.8.0 (527f934) arm64",
"date": "2024-01-07T14:25:53.000Z",
"device":"/dev/mmcblk0",
"addTime": "false",
"signature":"0x54 0x72",
"Transcend":"true",
"Secured mode": 0,
"Bus width": 4 bits
"Speed mode": Class 10
"UHS speed grade": 30MB/s and higher
"New bad blocks cnt": 0x00,
"Runtime spare blocks cnt": 0x33,
"Abnormal power loss": 0,
"Minimum erase cnt": 0,
"Maximum erase cnt": 1,
"Average erase cnt": 0,
"Remaining card life": 100"Total write CRC cnt": 0,
"Power cycle cnt": 0,
"NAND flash ID": 0x45,0x3c,0x98,0xb3,0x76,0x6b,
"IC": SM2707E,
"fw version": T0408 ,
"success":true
}

Reply:
Thank you for testing. The broken printing was due to not properly escaped % sign (which should be %%). This should be fixed now.